sdks/python/tox.ini

```diff
@@ -27,6 +27,7 @@ select = E3
 deps=
   pep8
   pylint
+  pyhamcrest
```

Comment:
My understanding of `setup.py` is that it defines how the Beam SDK gets installed, and thus also  dictates the list of dependencies installed on worker VMs to execute Beam pipelines. If that's the case, we should probably keep the dependency list as trim as possible.

@robertwb / @chamikaramj, can you shed light on whether this is the proper way to add a test dependency on `pyhamcrest`?